autofit/graphical/expectation_propagation/optimiser.py

```diff
@@ -155,12 +182,51 @@ def __init__(
             optimiser.paths = self.paths
 
         self.ep_history = ep_history or EPHistory()
+        self.factor_loggers = factor_loggers or {}
 
         with open(self.output_path / "graph.info", "w+") as f:
             f.write(self.factor_graph.info)
 
         self.visualiser = Visualise(self.ep_history, self.output_path)
 
+    @classmethod 
+    def from_meanfield(
+            cls,
+            model_approx: EPMeanField,
+            default_optimiser: Optional[AbstractFactorOptimiser] = None,
+            factor_optimisers: Optional[Dict[Factor, AbstractFactorOptimiser]] = None,
+            ep_history: Optional[EPHistory] = None,
+            factor_order: Optional[List[Factor]] = None,
+            paths: AbstractPaths = None,
+            factor_loggers: Optional[Dict[Factor, logging.Logger]] = None, 
+            exact_fit_kws=None, 
+    ):
+        factor_graph = model_approx.factor_graph 
+        factor_order = factor_order or factor_graph.factors
+
+        factor_optimisers = factor_optimisers or {}
+        factor_mean_field = model_approx.factor_mean_field
+        for factor in factor_graph.factors:
+            if factor not in factor_optimisers:
+                factor_dist = factor_mean_field[factor]
+                if factor.has_exact_projection(factor_dist):
+                    factor_optimisers[factor] = ExactFactorFit(**(exact_fit_kws or {}))
```

Comment:
It would probably make more sense to optionally pass in an ExactFactorFit rather than optionally pass in kwargs

Reply:
That potentially makes sense, I'm finding that I'm having to write custom optimisation loops, so I'm not using this class very much.